nodedb.go

```diff
@@ -151,13 +151,12 @@ func (ndb *nodeDB) GetNode(hash []byte) *Node {
 }
 
 func (ndb *nodeDB) GetFastNode(key []byte) (*FastNode, error) {
+	ndb.mtx.Lock()
+	defer ndb.mtx.Unlock()
```

Comment:
Can this be a RLock?

Reply:
wait why did moving this before `hasUpgradedToFastStorage` have any impact?

Reply:
`storageVersion` is still a common resource that might be written and read at the same time

In addition, for readability. Easier to see locks at the top of a method

Reply:
looking into `RLock`...

Reply:
[here](https://github.com/osmosis-labs/iavl/blob/93d75d2de14526b733cfbcca1faf92f98c3397cb/nodedb.go#L167) we are modifying a common resource. As a result, I don't think `RLock` could work